examples/md-flexible/src/Simulation.cpp

```diff
@@ -372,6 +465,106 @@ void Simulation::calculateGlobalForces(const std::array<double, 3> &globalForce)
   }
 }
 
+void Simulation::logSimulationState() {
+  int totalNumberOfParticles, ownedParticles, haloParticles;
+
+  int particleCount = _autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::ownedOrHalo);
+  autopas::AutoPas_MPI_Allreduce(&particleCount, &totalNumberOfParticles, 1, AUTOPAS_MPI_INT, AUTOPAS_MPI_SUM,
+                                 AUTOPAS_MPI_COMM_WORLD);
+
+  particleCount = _autoPasContainer->getNumberOfParticles();
+  autopas::AutoPas_MPI_Allreduce(&particleCount, &ownedParticles, 1, AUTOPAS_MPI_INT, AUTOPAS_MPI_SUM,
+                                 AUTOPAS_MPI_COMM_WORLD);
+
+  particleCount = _autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::halo);
+  autopas::AutoPas_MPI_Allreduce(&particleCount, &haloParticles, 1, AUTOPAS_MPI_INT, AUTOPAS_MPI_SUM,
+                                 AUTOPAS_MPI_COMM_WORLD);
+
+  double standardDeviationOfHomogeneity;
+  autopas::AutoPas_MPI_Allreduce(&_homogeneity, &standardDeviationOfHomogeneity, 1, AUTOPAS_MPI_DOUBLE, AUTOPAS_MPI_SUM,
+                                 AUTOPAS_MPI_COMM_WORLD);
+  standardDeviationOfHomogeneity = standardDeviationOfHomogeneity / _domainDecomposition.getNumberOfSubdomains();
```

Comment:
I think you have to do this a bit differently to be mathematically correct: https://stats.stackexchange.com/questions/25848/how-to-sum-a-standard-deviation

So sum the squares and then take the `sqrt()`